### DISCUSSION TITLE
你好，用了一年多skynet的mongodb存储很稳定，最近需要将mongodb连到mongodb+srv，使用云存储，我不太清楚如何处理，所以想请教下

### DISCUSSION THREAD

**Discussion by hujunlong** [cloudwu/skynet#1957] [General]

**Comment by JavenLaw** [comment]
不知道skynet与云存储（mongodb+srv）建立连接，是否与普通的mongodb实例连接相同？（但大概率是不同的了。。。）

不过据我所知，当你使用云存储（好像是Atlas?）的时候，进入控制面板，里面是可以选择driver的，可以有C/C++，Go等很多版本

你可以进去mongoDB官网，申请一个Atlas免费的体验账号，有500M的云存储

选择driver为C版本，用skynet的 mongo.client() 接口看能不能连接上

希望能帮到你

**Reply by hujunlong** [reply]
不一样，用aws创建的一个连接直接用nodejs就能访问到，skynet访问报socket错误

**Reply by ykxpb** [reply]
看起来是不支持 srv 协议，看文档这个协议需要查询 dns，获取 mongos 的地址列表和 option。你试试在创建 socketchannel 那里加上这个逻辑试试？

**Reply by ykxpb** [reply]
db/mongo.lua 创建socketchannel之前，根据 host 用 dns resolve下。

**Reply by ykxpb** [reply]
socket错误应该是说auth失败吧？因为配置的地址实际不是mongo的地址。

**Reply by mutou9267** [reply]
请教一下，我用mongodb，目前遇到一个问题，每过一段时间，db就会断掉，请问有心跳或者断掉重连的机制吗？不知道为什么就突然断掉了，谢谢。
